cognite/neat/graph/extractors/graph_sheet_to_graph.py

```diff
@@ -102,28 +102,58 @@ def sheet2triples(
                     )
 
                 # Adding object properties
-                elif class_property_pairs[sheet_name][property_].property_type == "ObjectProperty" and value:
-                    triples.extend(
-                        (
-                            instance_namespace[row.identifier],
-                            model_namespace[property_],
-                            instance_namespace[v.strip()],
+                elif value and class_property_pairs[sheet_name][property_].property_type == "ObjectProperty":
+                    if (
+                        class_property_pairs[sheet_name][property_].max_count
+                        and class_property_pairs[sheet_name][property_].max_count > 1
+                        and separator
+                    ):
+                        triples.extend(
+                            (
+                                instance_namespace[row.identifier],
+                                model_namespace[property_],
+                                instance_namespace[v.strip()],
+                            )
+                            for v in value.split(separator)
+                        )
+                    else:
+                        triples.append(
+                            (
+                                instance_namespace[row.identifier],
+                                model_namespace[property_],
+                                instance_namespace[value.strip()],
+                            )
```

Comment:
The nested conditionals becomes duplicated. Suggest rewriting this the conditionals in the loop as follows.
Note I renamed `property_` -> `property_name`. And use the `property_` variable for the property.
```python
            # iterate over sheet rows properties
            for property_name, value in row.to_dict().items():
                # Setting RDF type of the instance
                if property_name == "identifier":
                    triples.append(
                        (
                            instance_namespace[row.identifier],
                            RDF.type,
                            model_namespace[sheet_name],
                        )
                    )
                    continue
                elif not value:
                    continue

                property_ = class_property_pairs[sheet_name][property_name]
                is_one_to_many = (property_.max_count or 1) > 1 and separator
                values = value.split(separator) if is_one_to_many else [value]

                # Adding object properties
                if property_.property_type == "ObjectProperty":
                    triples.extend(
                        (
                            instance_namespace[row.identifier],
                            model_namespace[property_name],
                            instance_namespace[v.strip()],
                        )
                        for v in values
                    )
                # Adding data properties
                elif property_.property_type == "DatatypeProperty":
                    triples.extend(
                        (
                            instance_namespace[row.identifier],
                            model_namespace[property_name],
                            Literal(
                                v.strip(),
                                datatype=XSD[property_.expected_value_type],
                            ),
                        )
                        for v in values
                    )
```

Reply:
Consider what should happen if `not value` or whether there should be an `else` at the end of the loop. Do we expect other types than `ObjectProperty` or `DatatypeProperty`. If not this should raise an exception.

Reply:
added exception